scripts/docs-gen-snippets.mjs

```diff
@@ -0,0 +1,202 @@
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import yaml from 'js-yaml';
+
+const prettyAuthModes = {
+    OAUTH1: 'OAuth',
+    OAUTH2: 'OAuth',
+    OAUTH2_CC: 'OAuth',
+    BASIC: 'Basic',
+    API_KEY: 'API Key',
+    APP_STORE: 'Custom',
+    BILL: 'Bill',
+    SIGNATURE: 'Signature',
+    JWT: 'JWT',
+    TWO_STEP: 'Two Step',
+    TABLEAU: 'Tableau'
+};
+
+const providersPath = 'packages/shared/providers.yaml';
+const flowsPath = 'packages/shared/flows.yaml';
+const docsPath = 'docs-v2/integrations/all';
+const snippetsPath = 'docs-v2/snippets/generated';
+
+const providers = yaml.load(await fs.readFile(providersPath, 'utf-8'));
+const flows = yaml.load(await fs.readFile(flowsPath, 'utf-8'));
+
+const useCases = {};
+for (const [integration, config] of Object.entries(flows.integrations)) {
+    useCases[integration] = buildEndpoints(config.actions, integration).concat(buildEndpoints(config.syncs, integration));
+}
+
+const files = await fs.readdir(docsPath);
+for (const file of files) {
+    if (file.endsWith('.mdx')) {
+        const filePath = path.join(docsPath, file);
+        const content = await fs.readFile(filePath, 'utf-8');
+        let lines = content.split('\n');
+
+        // find the integration line
+        const providerLine = lines.find((line) => line.startsWith('provider: '));
+        if (!providerLine) {
+            throw new Error(`No provider line found in ${file}`);
+        }
+
+        const provider = providerLine.split('provider: ')[1].trim();
+
+        // write pre-built tooling snippet for the integration
+        const providerConfig = providers[provider];
+        if (!providerConfig) {
+            throw new Error(`Unknown provider ${provider} in ${file}`);
+        }
+        const authMode = providers[provider].auth_mode;
+        const prettyAuthMode = prettyAuthModes[authMode] || authMode;
+        const hasUseCases = useCases[provider] && useCases[provider].length > 0;
+        const toolingSnippet = preBuiltToolingSnippet({ prettyAuthMode, hasUseCases });
+
+        await fs.mkdir(`${snippetsPath}/${path.basename(file, '.mdx')}`, { recursive: true });
+        await fs.writeFile(`${snippetsPath}/${path.basename(file, '.mdx')}/PreBuiltTooling.mdx`, toolingSnippet, 'utf-8');
+
+        const casesSnippet = useCasesSnippet({ useCases: useCases[provider] });
+        await fs.writeFile(`${snippetsPath}/${path.basename(file, '.mdx')}/PreBuiltUseCases.mdx`, casesSnippet, 'utf-8');
```

Comment:
All the repeated path could be stored `const base = ${snippetsPath}/${path.basename(file, '.mdx')}`

Reply:
Fixed in 9c01c8d3. Good catch, thanks!